### PR TITLE
Update 3.7.7

### PR DESCRIPTION
**Pull request by Arishamays1**
> ## ☆Libarchive 3.7.7☆
> [Jira Ticket](https://anaconda.atlassian.net/browse/PKG-5931) 
[Upstream](https://github.com/libarchive/libarchive/tree/v3.7.7)
> 
> ### Changes
> * Updated version number and sha256
> * Updated dependencies for `run` and `host` 
> * Removed outdated `patches`

### CODE REVIEW

File: recipe/patches/0009-CMake-Force-Multi-threaded-DLL-runtime.patch [hunk 1]
```diff
@@ -1,34 +0,0 @@
...
-@@ -183,6 +185,8 @@ IF (MSVC)
-     SET(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} /wd4244")
-     SET(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} /wd4996")
-   ENDIF (MSVC_VERSION LESS 1900)
-+  # /MD option for Multi-threaded DLL.
-+  SET(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} /MD")
- ENDIF (MSVC)
- 
- # Enable CTest/CDash support
---
```
Comment (bwingconda) [inline recipe/patches/0009-CMake-Force-Multi-threaded-DLL-runtime.patch]: Enabling multi-threaded support still seems to be necessary for this version. Am I missing where this is being set elsewhere?

Comment (Arishamays1) [reply recipe/patches/0009-CMake-Force-Multi-threaded-DLL-runtime.patch]: Hi Brian, that was an oversight on my part! I’ll add the multi-threaded support back in. Thanks for catching that!

**Comment by cbouss** [comment]
Can you revert the changes to the patches? It seems line endings have been modified.

**Comment by cbouss** [comment]
I just saw this ticket related: https://anaconda.atlassian.net/browse/PKG-5733
Looking at the content of the produced package, the soname indeed changed from libarchive.so.20 in 3.7.4 to libarchive.so.13 in 3.7.7. Let's pause this for now.

**Comment by cbouss** [comment]
Or we could undo https://github.com/libarchive/libarchive/pull/1976 for this update. And wait for a 3.8 release to switch to the new soname.

**Comment by Arishamays1** [comment]
> I just saw this ticket related: https://anaconda.atlassian.net/browse/PKG-5733 Looking at the content of the produced package, the soname indeed changed from libarchive.so.20 in 3.7.4 to libarchive.so.13 in 3.7.7. Let's pause this for now.

This is fine! I can put this on hold for now and follow up on it if that works!

**Comment by JeanChristopheMorinPerso** [comment]
@cbouss however painful the ABI break might be, I think we need to do it now. I don't know when 3.8 will come out and it might not even happen.

This ABI breakage is a big problem that causes a bunch of issues between us and conda-forge which in itself breaks conda (or more libmamba, which breas conda-libmamba-solver which breaks conda).

I really think we should align with conda-forge here.

**Comment by cbouss** [comment]
ABI breakage addressed by https://github.com/AnacondaRecipes/repodata-hotfixes/pull/244

**Review by Jrice1317 — APPROVED** [review]

**Review by ELundby45 — APPROVED** [review]

**Comment by jezdez** [comment]
This isn't enough to prevent `conda update --all` to break conda environments

**Pull request merged**